dlib/test/pixel.cpp

```diff
@@ -753,6 +753,17 @@ namespace
         }
 
 
+        {
+            // bad pixel example, found through fuzzing
+            rgb_pixel src;
+            src.red=246;
+            src.green=170;
+            src.blue=245;
+            hsi_pixel dest;
+            assign_pixel(dest,src);
+            DLIB_TEST(dest.h>=0);
```

Comment:
These tests are always true, since `hsi_pixel` holds `unsigned char` typed fields.  How could this test have failed?

Reply:
It didn't, it's just there so the interesting event is actually triggered by unit test. Good for debugging, and good for keeping code coverage up and asserting what the function promises.

Reply:
I'm still confused about this.  Does something different happen if `DLIB_TEST(dest.h>=0);` is removed?

Reply:
Because it seems that it should be removed, as it can't fail and it suggests to the reader that somehow `dest.h` could be negative, but that is impossible.